openedx_events/learning/data.py

```diff
@@ -0,0 +1,89 @@
+"""
+Data attributes for events within the architecture subdomain `learning`.
+
+These attributes follow the form of attr objects specified in OEP-49 data
+pattern.
+"""
+from typing import Dict
+
+import attr
+from opaque_keys.edx.keys import CourseKey
+
+
+@attr.s(frozen=True)
+class UserProfileData:
+    """
+    Attributes defined for Open edX student's profile object.
+    """
+
+    meta = attr.ib(type=Dict[str, str], factory=dict)
+    name = attr.ib(type=str, factory=str)
+
+
+@attr.s(frozen=True)
+class StudentData:
+    """
+    Attributes defined for Open edX student object.
+    """
+
+    username = attr.ib(type=str)
+    email = attr.ib(type=str)
+    first_name = attr.ib(type=str, factory=str)
+    last_name = attr.ib(type=str, factory=str)
+    is_active = attr.ib(type=bool, default=True)
+    profile_meta = attr.ib(type=Dict[str, str], factory=dict)
+    profile_name = attr.ib(type=str, factory=str)
+
+
+@attr.s(frozen=True)
+class RegistrationData:
+    """
+    Attributes defined for Open edX registration profile object.
+    """
+
+    activation_key = attr.ib(type=str)
```

Comment:
This class was added to match [REGISTER_USER](https://github.com/eduNEXT/edx-platform/blob/384bc6b5147423f9c3208ce5c4afa79e5e0cd040/openedx/core/djangoapps/user_authn/views/register.py#L251) signal, but registration is only used in [these places](https://github.com/eduNEXT/edx-platform/blob/master/openedx/core/djangoapps/user_authn/views/register.py#L235-L240). I don't think `matching REGISTER_USER` is a sufficient argument to implement this class. I'll remove it.